### PR TITLE
Use `generate-changelog` action in `java-gradle-release`

### PR DESCRIPTION
**Pull request by MichaelKora**

### CODE REVIEW

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

**Review by yordanovsstoyan — DISMISSED** [review]

**Review by yordanovsstoyan — DISMISSED** [review]

File: .github/workflows/java-gradle-release.yaml [hunk 1]
```diff
@@ -8,6 +8,22 @@ on:
         description: "Scope of the release (major, minor or patch)."
         required: true
         type: string
+
```
Comment (torbsto) [inline .github/workflows/java-gradle-release.yaml:11]: Any reason for this empty line? There are none between all other parameters

File: actions/java-gradle-release/action.yaml [hunk 2]
```diff
@@ -2,6 +2,10 @@ name: "Release Java artifacts"
 description: "Release Java Gradle artifacts on Github"
 
 inputs:
+  changelog-file:
```
Comment (torbsto) [inline actions/java-gradle-release/action.yaml:5]: Missing documentation?

File: .github/workflows/java-gradle-release.yaml [hunk 3]
```diff
@@ -8,6 +8,22 @@ on:
...
+      changelog-file:
```
Comment (torbsto) [inline .github/workflows/java-gradle-release.yaml:12]: Missing documentation

File: actions/java-gradle-release/action.yaml [hunk 4]
```diff
@@ -84,19 +88,23 @@ runs:
...
 
-    - name: Commit and push
-      run: |
-        git add CHANGELOG.md
-        git commit -m "Changelog for version ${{ steps.evaluate-version.outputs.release-version }}" 
-        git push --follow-tags origin ${{ github.event.repository.default_branch }}
-      shell: bash
+    - name: Create changelog
+      id: build-changelog
+      uses: bakdata/ci-templates/actions/changelog-generate@chore/automatic-changelog
```
Comment (torbsto) [inline actions/java-gradle-release/action.yaml:95]: Please update

File: actions/java-gradle-release/action.yaml [hunk 5]
```diff
@@ -84,19 +88,23 @@ runs:
...
+      with:
+        github-token: ${{ inputs.github-token }}
+        new-tag: ${{ steps.evaluate-version.outputs.release-version }}
+        changelog-file: "${{ inputs.changelog-file }}"
+        fetch-reviewers: "true"
```
Comment (torbsto) [inline actions/java-gradle-release/action.yaml:100]: Why do we want this? Currently the changelog only contains the authors

**Review by torbsto — DISMISSED** [review]

File: actions/java-gradle-release-github/action.yaml [hunk 6]
```diff
@@ -48,13 +52,14 @@ runs:
         gradle-version: ${{ inputs.gradle-version }}
         gradle-cache: ${{ inputs.gradle-cache }}
```
Comment (yannick-roeder) [inline actions/java-gradle-release-github/action.yaml:53]: Java Gradle Setup is not needed anymore in this action then

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

File: actions/java-gradle-release/action.yaml [hunk 7]
```diff
@@ -86,17 +90,20 @@ runs:
...
+        github-token: ${{ inputs.github-token }}
+        new-tag: ${{ steps.evaluate-version.outputs.release-version }}
+        changelog-file: "${{ inputs.changelog-file }}"
+        fetch-release-information: "true"
 
-    - name: Commit and push
-      run: |
-        git add CHANGELOG.md
-        git commit -m "Changelog for version ${{ steps.evaluate-version.outputs.release-version }}"
-        git push --follow-tags origin ${{ github.event.repository.default_branch }}
```
[the remark this answers is not in the document]
Comment (yannick-roeder) [reply actions/java-gradle-release/action.yaml:101]: Remember that we only want to push the tag here, the GitHub release is created from CI running on the tag later

**Review by yannick-roeder — DISMISSED** [review]

File: .github/workflows/java-gradle-release.yaml [hunk 8]
```diff
@@ -62,9 +68,10 @@ jobs:
     steps:
       - name: Release on Github
         id: release
-        uses: bakdata/ci-templates/actions/java-gradle-release@1.27.0
+        uses: bakdata/ci-templates/actions/java-gradle-release@1.35.2
```
Comment (yannick-roeder) [inline .github/workflows/java-gradle-release.yaml:71]: This is not up to date anymore

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

**Review by yannick-roeder — APPROVED** [review]

**Pull request merged**